src/gt4py/gtscript.py

```diff
@@ -326,8 +326,12 @@ def lazy_stencil(
             Defers the generation step until the last moment and allows syntax checking independently.
             Also gives access to a more fine grained generate / build process.
     """
+    from gt4py import backend as backend_module
     from gt4py import frontend
 
+    if isinstance(backend, str):
+        backend = backend_module.from_name(backend)
```

Comment:
Does this fix a related problem?

Reply:
I will need to check that, this bit is very old.

Reply:
It didn't seem to make any tests fail when I removed it, let's wait for CI.

Reply:
Some tests failed now, so I will revert the removal.